src/Tests/Analyzers.Tests/RCS1077OptimizeLinqMethodCallTests.cs

```diff
@@ -959,6 +959,46 @@ void M()
 ");
     }
 
+    [Fact, Trait(Traits.Analyzer, DiagnosticIdentifiers.OptimizeLinqMethodCall)]
+    public async Task Test_CallOrderInsteadOfOrderByIdentity()
+    {
+        const string input = @"
+using System.Collections.Generic;
+using System.Linq;
+
+class C
+{
+    void M()
+    {
+        IEnumerable<object> x = null;
+
+        x = x.[|OrderBy(f => f)|];
+    }
+}
+";
+        // Order() is available in .NET 7 and greater
+        // https://learn.microsoft.com/en-us/dotnet/api/system.linq.enumerable.order?view=net-8.0&viewFallbackFrom=net-6.0
+#if NET7_0_OR_GREATER
+        const string fix = @"
+using System.Collections.Generic;
+using System.Linq;
+
+class C
+{
+    void M()
+    {
+        IEnumerable<object> x = null;
+
+        x = x.Order();
+    }
+}
+";
+        await VerifyDiagnosticAndFixAsync(input, fix);
+#else
+    await VerifyNoDiagnosticAsync(input);
+#endif
```

Comment:
I created a PR which update target framework to .NET 8: https://github.com/dotnet/roslynator/pull/1523.

After that PR is merged and this branch is updated the test should pass without the conditional compilation.

Reply:
@BenjaminBrienen I've merged the PR so you can update this branch.

Reply:
Nice. Those other mistakes would've been caught with the updated target framework. Should I remove the conditional compilation and wait to merge until #1523 is merged?

Reply:
I've updated the branch so now the tests should pass.

Reply:
After I fix the ArgumentsList, yeah. Currently trying to figure out how to remove arguments. At the moment, I only see methods for adding arguments and ArgumentsList doesn't have a public constructor.